### PR TITLE
Feat: allow comments to be attached for identifiers used in definitions

### PR DESCRIPTION
**Pull request by georgesittas**
Context: https://tobiko-data.slack.com/archives/C0448SFS3PF/p1713777996852009

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -5517,16 +5518,18 @@ def _parse_id_var(
...
-
-        if (any_token and self._advance_any()) or self._match_set(tokens or self.ID_VAR_TOKENS):
+        expression = self._parse_identifier()
+        if not expression and (
+            (any_token and self._advance_any()) or self._match_set(tokens or self.ID_VAR_TOKENS)
+        ):
             quoted = self._prev.token_type == TokenType.STRING
-            return exp.Identifier(this=self._prev.text, quoted=quoted)
+            expression = exp.Identifier(this=self._prev.text, quoted=quoted)
+            if comment:
```
Comment (tobymao) [inline sqlglot/parser.py:5529]: why is this needed? why can't we just always add comments?

Comment (tobymao) [reply sqlglot/parser.py:5529]: self.expression(exp.Identifier, ...)

Comment (georgesittas) [reply sqlglot/parser.py:5529]: Because the comments associated with identifiers are actually attached in columns, due to the `self.expression(exp.Column, ...)` call, which in turn calls `self._add_comments`.

File: sqlglot/parser.py [hunk 2]
```diff
@@ -5517,16 +5518,18 @@ def _parse_id_var(
...
+                self._add_comments(expression)
```
[the remark this answers is not in the document]
Comment (georgesittas) [reply sqlglot/parser.py:5530]: It's handled automatically because `_parse_identifier` constructs an expression using `self.expression`, which in turn takes care of attaching comments:

```python
return self.expression(exp.Identifier, this=self._prev.text, quoted=True)
```

File: sqlglot/parser.py (hunk 1 above)
Comment (georgesittas) [reply sqlglot/parser.py:5529]: So if you prefer moving comments at the identifier level (which is a breaking change), then your suggestion would work fine. Perhaps we could route `Column.comments` to `Column.this.comments` or something.

Comment (georgesittas) [reply sqlglot/parser.py:5529]: FWIW the behavior is now inconsistent because delimited identifiers do get comments attached to them, as I explained in https://github.com/tobymao/sqlglot/pull/3340#discussion_r1575032887. So perhaps we should either change that or make sure we also do it for non-delimited ones.

File: sqlglot/parser.py [hunk 3]
```diff
@@ -5503,13 +5506,13 @@ def _parse_alias(
         )
 
         if alias:
+            comments += alias.pop_comments()
```
Comment (tobymao) [inline sqlglot/parser.py:5509]: why += instead of extends?

File: tests/test_parser.py [hunk 4]
```diff
@@ -357,7 +357,7 @@ def test_comments_insert(self):
         )
 
         self.assertEqual(expression.comments, ["comment1.1", "comment1.2", "comment1.3"])
-        self.assertEqual(expression.this.comments, ["comment2"])
+        self.assertEqual(expression.this.this.comments, ["comment2"])
```
Comment (tobymao) [inline tests/test_parser.py:360]: why did this change? don't we push the comments up?

Comment (georgesittas) [reply tests/test_parser.py:360]: Because this corresponds to a `Table`. May need to push the comments to tables as well.

Comment (georgesittas) [reply tests/test_parser.py]: [done](https://github.com/tobymao/sqlglot/pull/3340/commits/47a57456d665a2ad65f5a80b0575f000abfcec43)

**Review by tobymao — APPROVED** [review]
